File: app/src/main/java/pl/srw/billcalculator/settings/di/SettingsComponent.java
package pl.srw.billcalculator.settings.di;

import dagger.Subcomponent;
import pl.srw.billcalculator.settings.activity.SettingsActivity;
import pl.srw.billcalculator.settings.fragment.PgeSettingsFragment;
import pl.srw.billcalculator.settings.fragment.PgnigSettingsFragment;
import pl.srw.billcalculator.settings.fragment.TauronSettingsFragment;
import pl.srw.billcalculator.settings.restore.ConfirmRestoreSettingsDialogFragment;
import pl.srw.mfvp.di.scope.RetainActivityScope;

@RetainActivityScope
@Subcomponent
public interface SettingsComponent extends ConfirmRestoreSettingsComponentInjectable<SettingsActivity> {

    void inject(ConfirmRestoreSettingsDialogFragment fragment);

    void inject(PgeSettingsFragment pgeSettingsFragment);

    void inject(PgnigSettingsFragment pgnigSettingsFragment);

    void inject(TauronSettingsFragment tauronSettingsFragment);
}

File: app/src/main/java/pl/srw/billcalculator/settings/di/SettingsComponent.kt
package pl.srw.billcalculator.settings.di

import dagger.Subcomponent
import pl.srw.billcalculator.settings.SettingsActivity
import pl.srw.billcalculator.settings.details.SettingsDetailsFragment
import pl.srw.billcalculator.settings.details.dialog.InputSettingsDialogFragment
import pl.srw.billcalculator.settings.details.dialog.PickingSettingsDialogFragment
import pl.srw.billcalculator.settings.details.restore.ConfirmRestoreSettingsDialogFragment
import pl.srw.billcalculator.settings.list.SettingsFragment
import pl.srw.mfvp.di.scope.RetainActivityScope

@RetainActivityScope
@Subcomponent
interface SettingsComponent {

    fun inject(activity: SettingsActivity)

    fun inject(fragment: SettingsFragment)

    fun inject(fragment: SettingsDetailsFragment)

    fun inject(dialog: ConfirmRestoreSettingsDialogFragment)

    fun inject(dialog: InputSettingsDialogFragment)

    fun inject(dialog: PickingSettingsDialogFragment)
}
